example-plugins/example_patient_sync/protocols/patient_sync.py

```diff
@@ -0,0 +1,155 @@
+from typing import Any
+
+from canvas_sdk.effects import Effect
+from canvas_sdk.effects.patient import CreatePatientExternalIdentifier
+from canvas_sdk.events import EventType
+from canvas_sdk.protocols import BaseProtocol
+from canvas_sdk.utils import Http
+from canvas_sdk.v1.data.patient import Patient
+from logger import log
+
+PARTNER_URL_BASE = "https://app.usebridge.xyz"
+SYSTEM_URL = "https://canvas.app.usebridge.xyz"
+
+
+class PatientSync(BaseProtocol):
+    """Protocol for synchronizing patient data between systems."""
+
+    RESPONDS_TO = [
+        EventType.Name(EventType.PATIENT_CREATED),
+    ]
+
+    @property
+    def partner_api_base_url(self) -> str:
+        """Return the base URL for the external partner API."""
+        return self.sanitize_url(self.secrets["PARTNER_API_BASE_URL"] or f"{PARTNER_URL_BASE}/api")
+
+    @property
+    def partner_ui_base_url(self) -> str:
+        """Return the base URL for the external partner UI."""
+        return self.sanitize_url(self.secrets["PARTNER_UI_BASE_URL"] or PARTNER_URL_BASE)
+
+    @property
+    def partner_request_headers(self) -> dict[str, str]:
+        """Return the request headers for external partner API requests."""
+        partner_secret_api_key = self.secrets["PARTNER_SECRET_API_KEY"]
+        return {"X-API-Key": partner_secret_api_key}
+
+    @property
+    def partner_patient_metadata(self) -> Any:
+        """Return metadata for creation of the patient on external partner platform."""
+        metadata = {"canvasPatientId": self.target}
+
+        canvas_url = self.secrets["CANVAS_BASE_URL"]
```

Comment:
The Canvas base URL can be determined using `self.environment`.

```python
subdomain = self.environment["CUSTOMER_IDENTIFIER"]
canvas_url = f"https://{subdomain}.canvasmedical.com"
```

Reply:
oooh! TIL